lib/error.js

```diff
@@ -232,4 +232,36 @@ AppError.expiredToken = function expiredToken(expiredAt) {
   });
 };
 
+AppError.notPublicClient = function unknownClient(clientId) {
+  return new AppError({
+    code: 400,
+    error: 'Bad Request',
+    errno: 116,
+    message: 'Not a public client'
+  }, {
+    clientId: clientId
+  });
+};
+
+
+AppError.mismatchCodeChallenge = function mismatchCodeChallenge(code_challenge) {
+  return new AppError({
+    code: 400,
+    error: 'Bad Request',
+    errno: 117,
+    message: 'Incorrect code_challenge'
+  }, {
+    requestCodeChallenge: code_challenge
+  });
+};
+
+AppError.missingPkceParameters = function invalidScopes() {
```

Comment:
Function name of `invalidScopes` doesn't match the error. Likely a left over from copying a previous error constructor.